src/libratbag-hidraw.c

```diff
@@ -1660,3 +1660,49 @@ ratbag_hidraw_read_input_report_index(struct ratbag_device *device, uint8_t *buf
 
 	return rc >= 0 ? rc : -errno;
 }
+
+int
+ratbag_hidraw_read_input_report_filter(struct ratbag_device *device, uint8_t *buf, size_t len, int *(*filter)(uint8_t *buf, size_t len))
+{
+	return ratbag_hidraw_read_input_report_index_filter(device, buf, len, 0, filter);
+}
+
+int
+ratbag_hidraw_read_input_report_index_filter(
```

Comment:
same goes for the function above, I'm wondering if `filtered` is clearer for a name, i.e.
`ratbag_hidraw_read_input_report_index_filtered`